algo/src/main/java/com/hartwig/actin/algo/evaluation/complication/PatternMatcher.java

```diff
@@ -30,4 +35,37 @@ public static boolean isMatch(@NotNull String term, @NotNull Set<List<String>> p
 
         return false;
     }
+
+    public static Set<String> findComplicationNamesMatchingCategories(PatientRecord record,
```

Comment:
I would suggest a name that makes it more clear that "any" category match is allowed. The name of the function could be interpreted as "all". Maybe "findComplicationNamesMatchingAtLeastOneCategory". Or potentially "Any" instead of "AtLeastOne" to be less verbose?